spec/models/conversation_spec.rb

```diff
@@ -9,21 +9,15 @@
     it { expect(subject).to have_many(:conversations_users) }
   end
 
-  context '#create_conversation' do
+  context 'validate uniqueness between users' do
     let(:user1) { create(:user) }
     let(:user2) { create(:user) }
 
-    context 'when conversation between users does not exsist' do
-      it 'creates a conversation' do
-        expect { Conversation.create_conversation(user1, user2) }.to change { Conversation.count }.by(1)
-      end
-    end
+    let!(:conversation) { create(:conversation, :with_users, user1: user1, user2: user2) }
+    subject { create(:conversation, :with_users, user1: user2, user2: user1) }
 
-    context 'when conversation between users already exsist' do
-      let!(:conversation) { Conversation.create_conversation(user1, user2) }
-      it 'does not creates a conversation' do
-        expect { Conversation.create_conversation(user2, user1) }.to change { Conversation.count }.by(0)
-      end
+    it 'not to be valid' do
+      expect { subject }.to raise_error(ActiveRecord::RecordInvalid)
```

Comment:
I think you should catch the error from active record in the controller and return an error defined in the locale yml.
WDYT @GuilleLeopold ?

Reply:
Maybe thats an option. But, from my point of view, thats not necessary. Because a conversation is never created from an endpoint directly.

The only way to create a new conversation and raise that error is using the endpoint to create a new target (which has compatibles target). That endpoint, as far as i know, it should not return an error if the conversation is not created.